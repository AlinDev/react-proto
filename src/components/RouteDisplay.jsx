import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';
import PropTypes from 'prop-types';

const RouteDisplay = ({
  componentTitle,
  pathName,
  routeCompId,
  deleteRoute,
  classes,
  routerCompId,
}) => (
  <div className="routeListItem">
    Path: / {pathName}
    Component: {componentTitle}
    <IconButton
      className={classes.button}
      onClick={() => {
        deleteRoute({
          routeCompId,
          routerCompId,
        });
      }}
      aria-label='Delete'>
      <DeleteIcon className={classes.light} />
    </IconButton>
  </div>
);

export default RouteDisplay;

RouteDisplay.propTypes = {
  deleteRoute: PropTypes.func.isRequired,
  routerCompId: PropTypes.string.isRequired,
  componentTitle: PropTypes.string.isRequired,
  pathName: PropTypes.string.isRequired,
  routeCompId: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
};
