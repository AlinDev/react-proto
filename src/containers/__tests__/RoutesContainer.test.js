import React, { Component, Children } from 'react';
import renderer from 'react-test-renderer';
import TextField from '@material-ui/core/TextField';
import { shallow, mount } from 'enzyme';
import Select from '@material-ui/core/Select';
import Button from '@material-ui/core/Button';
import ConnectedRoutesContainer, { RoutesContainer } from '../RoutesContainer.jsx';
import { styles } from '../LeftComponentContainer.jsx';
import RouteDisplay from '../../components/RouteDisplay.jsx';

describe('Routes Container Tests', () => {
  let wrapper;
  const addRoute = jest.fn();

  const props = {
    classes: {},
    addRoute,
    deleteRoute: () => null,
    setSelectableRoutes: () => null,
    setVisible: () => null,
    component: {
      id: '3',
      selectableRoutes: [
        {
          id: '1',
          title: 'route1',
        },
        {
          id: '2',
          title: 'route2',
        },
      ],
      routes: [
        {
          routeCompId: '3',
          title: 'route3',
        },
        {
          routeCompId: '4',
          title: 'route4',
        },
      ],
    },
  };

  beforeAll(() => {
    wrapper = mount(<RoutesContainer {...props} />);
  });

  it('snapshot test', () => {
    const tree = renderer.create(<RoutesContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an input field that calls handleUpdate when changed', () => {
    const inputField = wrapper.find(TextField);
    expect(inputField).toHaveLength(1);
    const instance = wrapper.instance();

    expect(inputField.prop('onChange')).toBeDefined();
    expect(inputField.prop('onChange')).toBeInstanceOf(Function);
    expect(inputField.prop('onChange')).toEqual(instance.handleChange);
  });

  it('should have a handleUpdate function which sets the state to the current string', () => {
    expect(wrapper.state().pathName).toBe('');
    wrapper.find(TextField).prop('onChange')({ target: { value: 'a' }, preventDefault: () => null });
    expect(wrapper.state().pathName).toBe('a');
  });

  it('should contain two RoutesContainer MUI components', () => {
    expect(wrapper.find(RouteDisplay)).toHaveLength(2);
  });

  it('should contain three route child options', () => {
    const optionslist = wrapper.find('option');
    expect(optionslist).toHaveLength(3);
    expect(optionslist.at(0).text()).toBe('None');
    expect(optionslist.at(1).text()).toBe('route1');
    expect(optionslist.at(2).text()).toBe('route2');
  });

  it('should set the selectedRouteId in state when routeChild is selected', () => {
    const selectComp = wrapper.find('Select');
    selectComp.prop('onChange')({ target: { value: '2' } });
    expect(wrapper.state().selectedRouteId).toBe('2');
  });

  it('should run handleAddRoute when the add button is clicked', () => {
    const instance = wrapper.instance();
    const buttonClickHandler = wrapper.find(Button).prop('onClick');
    expect(buttonClickHandler).toEqual(instance.handleAddRoute);
  });

  it('should pass route object into AddRoute when the add button is clicked', () => {
    wrapper.find(Button).prop('onClick')();
    expect(addRoute).toBeCalledWith({ path: 'a', routeCompId: '2', routerCompId: '3' });
  });

  it('should contain two RoutesContainer MUI components', () => {
    expect(wrapper.find(RouteDisplay)).toHaveLength(2);
  });
});
