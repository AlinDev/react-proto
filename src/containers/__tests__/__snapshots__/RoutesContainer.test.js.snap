// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Container Tests snapshot test 1`] = `
<div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  >
    <div
      align="stretch"
      className="MuiGrid-container-5 MuiGrid-align-items-xs-baseline-16"
    >
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-10-43"
      >
        <label
          className="MuiFormLabel-root-107 MuiInputLabel-root-102 MuiInputLabel-formControl-103 MuiInputLabel-animated-106"
          data-shrink={false}
          htmlFor="stateful"
        >
          Path: /
        </label>
      </div>
      <br />
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-10-43"
      >
        <div
          className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
        >
          <div
            className="MuiInput-root-114 MuiInput-formControl-115 MuiInput-underline-118"
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiInput-input-122"
              disabled={false}
              id="title-input"
              name="pathName"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="RouteName"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-2-35"
      >
        <button
          aria-label="Add"
          className="MuiButtonBase-root-153 MuiButton-root-127 MuiButton-fab-144 MuiButton-mini-149 MuiButton-contained-138 MuiButton-containedPrimary-139 MuiButton-raised-141 MuiButton-raisedPrimary-142"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-128"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-156"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-163"
          />
        </button>
      </div>
    </div>
  </div>
  <div>
    <label
      className="MuiFormLabel-root-107 MuiInputLabel-root-102 MuiInputLabel-animated-106"
      htmlFor="parentSelect"
    >
      Route Child
    </label>
    <div
      className="MuiInput-root-114 MuiInput-underline-118"
    >
      <div
        className="MuiSelect-root-170"
      >
        <select
          className="MuiSelect-select-171 MuiInput-input-122"
          id="componentSelect"
          name="componentName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
        >
          <option
            value=""
          >
            None
          </option>
          <option
            value="1"
          >
            route1
          </option>
          <option
            value="2"
          >
            route2
          </option>
        </select>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-156 MuiSelect-icon-174"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <br />
  <div>
    Routes:
    <br />
    <div
      align="stretch"
      className="MuiGrid-container-5 MuiGrid-align-items-xs-baseline-16"
    >
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-10-43"
      >
        Path: / 
        <br />
        Component: 
      </div>
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-2-35"
      >
        <button
          aria-label="Delete"
          className="MuiButtonBase-root-153 MuiIconButton-root-175"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-180"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-156"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-163"
          />
        </button>
      </div>
    </div>
    <div
      align="stretch"
      className="MuiGrid-container-5 MuiGrid-align-items-xs-baseline-16"
    >
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-10-43"
      >
        Path: / 
        <br />
        Component: 
      </div>
      <div
        className="MuiGrid-item-6 MuiGrid-grid-xs-2-35"
      >
        <button
          aria-label="Delete"
          className="MuiButtonBase-root-153 MuiIconButton-root-175"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-180"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-156"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-163"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
